sentry-ruby/lib/sentry/span.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+require "securerandom"
+
+module Sentry
+  class Span
+    STATUS_MAP = {
+      400 => "invalid_argument",
+      401 => "unauthenticated",
+      403 => "permission_denied",
+      404 => "not_found",
+      409 => "already_exists",
+      429 => "resource_exhausted",
+      499 => "cancelled",
+      500 => "internal_error",
+      501 => "unimplemented",
+      503 => "unavailable",
+      504 => "deadline_exceeded"
+    }
+
+
+    attr_reader :trace_id, :span_id, :parent_span_id, :sampled, :start_timestamp, :timestamp, :description, :op, :status, :tags, :data
+    attr_accessor :span_recorder
+
+    def initialize(description: nil, op: nil, status: nil, trace_id: nil, parent_span_id: nil, sampled: nil, start_timestamp: nil, timestamp: nil)
+      @trace_id = trace_id || SecureRandom.uuid.delete("-")
+      @span_id = SecureRandom.hex(8)
+      @parent_span_id = parent_span_id
+      @sampled = sampled
+      @start_timestamp = start_timestamp || Time.now.utc.to_f
+      @timestamp = timestamp
+      @description = description
+      @op = op
+      @status = status
+      @data = {}
+      @tags = {}
+      @span_recorder = SpanRecorder.new(1000)
+      @span_recorder.add(self)
+    end
+
+    def finish
+      # already finished
+      return if @timestamp
+
+      @timestamp = Time.now.utc.to_f
+    end
+
+    def to_sentry_trace
+      sampled_flag = ""
+      sampled_flag = @sampled ? 1 : 0 unless @sampled.nil?
+
+      "#{@trace_id}-#{@span_id}-#{sampled_flag}"
```

Comment:
I think here is a nit, if `sampled` is nil which is a valid value we would emit `#{@trace_id}-#{@span_id}-` with a trailing `-`

we should change that (eventho I think it would still work)

Reply:
if `@samped` is `nil`, `sampled_flag` will be an empty string, which will add nothing to the string and become

```
"#{@trace_id}-#{@span_id}-"
```

so exactly the expected behavior 😄 

here's the spec for such case https://github.com/getsentry/sentry-ruby/pull/1108/files#diff-a54e868d59f29d1f5369002784782b73b90c18c6493aab761e0230a51329fb6bR52-R61